code/game/mecha/equipment/tools/work_tools.dm

```diff
@@ -30,28 +30,39 @@
 		return
 	if(!cargo_holder)
 		return
-	if(istype(target,/obj))
+	if(istype(target, /obj))
 		var/obj/O = target
-		if(!O.anchored)
-			if(cargo_holder.cargo.len < cargo_holder.cargo_capacity)
-				chassis.visible_message("[chassis] lifts [target] and starts to load it into cargo compartment.")
-				O.anchored = 1
-				if(do_after_cooldown(target))
-					cargo_holder.cargo += O
-					O.loc = chassis
-					O.anchored = 0
-					occupant_message("<span class='notice'>[target] successfully loaded.</span>")
-					log_message("Loaded [O]. Cargo compartment capacity: [cargo_holder.cargo_capacity - cargo_holder.cargo.len]")
-				else
-					O.anchored = initial(O.anchored)
-			else
-				occupant_message("<span class='warning'>Not enough room in cargo compartment!</span>")
-		else
+		if(istype(target, /obj/machinery/power/supermatter_crystal)) //No, you can't pick up the SM with this you moron, did you think you were clever?
+			var/obj/machinery/power/supermatter_crystal/SM = target
+			var/obj/mecha/working/ripley/R = chassis
+			if(R.cargo)
+				R.cargo.Cut() //We don't want to drop cargo that just spam hits the SM, let's delete it
```

Comment:
I can't check right now but, If `cargo` contains actual references to objects, you probably need to do `QDEL_LIST(R.cargo)`

Reply:
Hmm, it seemed to work when I tested it, cut just removes the list contents here.

Reply:
It's just that if `cargo` has a bunch of references, and you just cut the list's contents, those references will be floating around. Or probably in the `contents` of the cargo container. So you might need to qdel them, if the mech getting destroyed doesn't already handle that. I'll have to check this out when I get home later, I don't have the code with me.

Reply:
The references wont be "Floating around". If you cut the list you remove the references. 

If you were to use `QDEL_LIST`, that would delete the objects that are in the list themeslves entirely, not just remove the list entries.

Use `R.cargo.Cut()`.

Reply:
Everything in the list should also be deleted, because everything in the mech should be destroyed along with it, so I think QDEL_LIST is fine here.

Reply:
"Floating around" is definitely wrong, I'm not sure what I meant by that. But yeah I was going by the comment and how it said the cargo should be deleted, which is why I suggested `QDEL_LIST`.